### DISCUSSION TITLE
Coredump when running `list-fonts`

### DISCUSSION THREAD

**Discussion by rrrnld** [kovidgoyal/kitty#7896] [Q&A]
I somehow managed to get my system into a state where kitty crashes when I run `kitty list-fonts`. I guess this is due to a specific font that kitty has problems with. `kitty list-fonts`  loads the list of fonts, but scrolling around will cause `` eventually. Another message I have seen is `Error: signal: segmentation fault (core dumped)`.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x8a1aa7]

goroutine 47 [running]:
kitty/kittens/choose_fonts.(*kitty_font_backend_type).send.func1()
	kitty/kittens/choose_fonts/backend.go:74 +0x47
created by kitty/kittens/choose_fonts.(*kitty_font_backend_type).send in goroutine 52
	kitty/kittens/choose_fonts/backend.go:73 +0x1ba
```

How can I figure out what causes the crash? Thanks!

PS: This is on kitty 0.36.1 on NixOS.

**Comment by kovidgoyal** [comment]
well that particular crash is fixed by:
https://github.com/kovidgoyal/kitty/commit/0a7812ab20ec529e5a96c75bbcd1c58a3e25ae24

and note that the choose-fonts kitten crashing doesnt affect the rest of
kitty.  Note which font is causing the crash and attach it here. Also
specify what OS etc you are running on.

**Reply by rrrnld** [reply]
Any idea how to best find out which font is causing the crash?

**Comment by kovidgoyal** [comment]
On Wed, Sep 25, 2024 at 02:34:44PM -0700, rrrnld wrote:
> Any idea how to best find out which font is causing the crash?

Scroll down the list of fonts till you reach one that triggers a crash.

**Reply by rrrnld** [reply]
I'm not exactly sure what happens, and honestly it's absolutely cursed. My text editor crashed as well, so I'm not sure it's something that kitty should be able to handle. Something went wrong when installing the fonts and I'm afraid it's faulty hardware.

**Reply by kovidgoyal** [reply]
See #7892
